main/src/ca/uwaterloo/flix/language/phase/Simplifier.scala

```diff
@@ -78,60 +90,74 @@ object Simplifier {
             // Translate to InvokeMethod exp
             val strClass = Class.forName("java.lang.String")
             val method = strClass.getMethod("concat", strClass)
-            SimplifiedAst.Expr.ApplyAtomic(AtomicOp.InvokeMethod(method), es, tpe, purity, loc)
+            val t = visitType(tpe)
+            SimplifiedAst.Expr.ApplyAtomic(AtomicOp.InvokeMethod(method), es, t, purity, loc)
 
           case AtomicOp.ArrayLit | AtomicOp.ArrayNew =>
             // The region expression is dropped (head of exps / es)
             val es1 = es.tail
-            SimplifiedAst.Expr.ApplyAtomic(op, es1, tpe, purity, loc)
+            val t = visitType(tpe)
+            SimplifiedAst.Expr.ApplyAtomic(op, es1, t, purity, loc)
 
           case AtomicOp.Ref =>
             // The region expression is dropped (tail of exps / es)
             val es1 = List(es.head)
-            SimplifiedAst.Expr.ApplyAtomic(op, es1, tpe, purity, loc)
+            val t = visitType(tpe)
+            SimplifiedAst.Expr.ApplyAtomic(op, es1, t, purity, loc)
 
           case AtomicOp.Spawn =>
-            // Wrap the expression in a closure: () -> tpe \ eff
+            // Wrap the expression in a closure: () -> tpe \ ef
             val List(e1, e2) = es
-            val lambdaTyp = Type.mkArrowWithEffect(Type.Unit, eff, e1.tpe, loc)
-            val fp = SimplifiedAst.FormalParam(Symbol.freshVarSym("_spawn", BoundBy.FormalParam, loc), Ast.Modifiers.Empty, Type.mkUnit(loc), loc)
+            val lambdaTyp = MonoType.Arrow(Nil, e1.tpe)
```

Comment:
I think here `Nil` should be `List(Unit)`.

Reply:
Both pass the tests

Reply:
Yes, but List(Unit) is more correct.